### ISSUE TITLE
Siri remote and keyboard differences with yab

### ISSUE THREAD

**Issue by kambala-decapitator** [SylvainCecchetto/xbmc#62]
as reported by Bry Jacks in Slack:
1. when you swipe on apple remove you should get fast movement (right now that works but swiping and holding)
2. and subtle taps in the top bottom left right areas should give you slow movement
3. keyboard used to come up with apples keyboard (including voice) but now its kodi’s keyboard
4. another problem I’m seeing with remote/keyboard is that when using the apple remote on an iPhone - for some reason when in a text box the iphone remote doesn’t know you are in text so it doesn’t bring up the phone keyboard

**Comment by kambala-decapitator** [comment]
Siri remote config: `system/keymaps/customcontroller.SiriRemote.xml`. You might want to check custom remote config as well: https://github.com/kambala-decapitator/xbmc/blob/ATV-working/system/keymaps/customcontroller.SiriRemote.xml

as for keyboard, it's `xbmc/platform/darwin/ios-common/IOSKeyboard.mm` I guess.

**Comment by sasagr** [comment]
If you are interested on an old user opinion, myself I don't displease the kodi keyboard especially if the current autocompletion addon is proved to be compatible with it. Of course if the voice recognition could be added to it, it would be perfect. Otherwise what about leaving the two options (kodi and tvO keyboards) and decide the user to decide from the settings

**Comment by sy6sy2** [comment]
@kambala-decapitator  Concerning point 3. I already open a PR in order to restore vanilla tvOS keyboard and Fuzzard merged it, but I don't know why but my commit is not on `fuzzard:master_darwin-embedded-ATV` anymore.
The only thing to restore tvOS keyboard is to cherry pick this commit https://github.com/fuzzard/xbmc/pull/3/commits/fe9a2f81d4646436f031cddec742fba3bcb345ba.

@sasagr  I personally prefer tvOS keyboard because I can use Siri remote microphone and also my iPhone to send text ;-) But it should be possible to add a setting in order to choose between tvOS or Kodi keyboard ;-)

**Comment by sasagr** [comment]
@SylvainCecchetto pls let me know if this option of having both keyboard in settings can and will be introduced. I really hate the native tvOS keyboard layout and my English accent is not really so perfect sometimes :)

**Comment by sy6sy2** [comment]
I will have a look to add a setting in order to choose between native or Kodi keyboard.
@fuzzard just restored on `master_darwin-embedded-ATV` the native keyboard.

**Comment by sasagr** [comment]
Yes. I noticed about the native keyboard being restored. I will keep an eye on your next changes. Thanks

**Comment by sy6sy2** [comment]
@sasagr  Would you try this? https://github.com/fuzzard/xbmc/pull/6
I added a setting in order to switch between tvOS or Kodi virtual keyboard.

@kambala-decapitator  Maybe with the correct settings restored by my PR you can test point 1 and 2?

![Screen Shot 2019-05-18 at 17 14 15](https://user-images.githubusercontent.com/11562279/57971604-7499d180-7990-11e9-9e24-39ca3e9ec772.png)

**Comment by sasagr** [comment]
@SylvainCecchetto I would like to do that but since I m honestly not so skilled with trying from commits pls let me ask you what is the shortest way to do so. My idea was to clone again the actual branch and replace the 5 files you changed, using GitHub Desktop maybe, and rebuild everything from there. It will take me 5-6 hours at least.

**Comment by sy6sy2** [comment]
You can wait until fuzzard merge the pull request. Then you will only have to do a `git pull` and re-build Kodi ;-)

**Comment by sasagr** [comment]
I see it has been merged. I will pull and build it again. do I need to Build tools and dependencies again? and also binary addons?
 It is a painful job ;) 
maybe I can just git pull on top of what I have already and directly build Kodi as per 6. of doc instructions?

**Comment by sasagr** [comment]
all done! Works as expected. Thanks! will let u know if any issues in the next days/test

**Comment by sy6sy2** [comment]
Sorry I am a little late.
Glad that is work for you, thank you for your feedback ;-)

**Comment by phunkyfish** [comment]
So on the fuzzard branch this is still an issue:

- when you swipe on apple remove you should get fast movement (right now that works but swiping and holding)

Can someone else test the the version in the fuzzard branch:

https://github.com/fuzzard/xbmc/blob/master_darwin-embedded-ATV/system/keymaps/customcontroller.SiriRemote.xml

Against:

https://github.com/kambala-decapitator/xbmc/blob/ATV-working/system/keymaps/customcontroller.SiriRemote.xml

I will test this evening but as I don't use the Siri remote regularly I'm not sure I would notice the difference.

**Comment by sy6sy2** [comment]
The "fast movement" works for me when you enable "Match Apple tvOS standard (Siri Remote)" in input settings.

**Comment by phunkyfish** [comment]
Ah, I must have a slightly older build. Does this use a different keymap or the same one that’s in the fuzzard branch?

I’ve noticed some issues using a CEC remote (super fast movement if you hold directional buttons on the remote), and wondering if they are related.

**Comment by sy6sy2** [comment]
It's the one from fuzzard branch, I never change the keymap.
Related commit is here: https://github.com/fuzzard/xbmc/commit/d1ee7102243758458a8c6d26a2afa7ec2adcd84b

I don't know :-/ I am not using CEC remote.

**Comment by phunkyfish** [comment]
So the options in input settings for apple remote are Disabled, Standard, Universal and Harmony. I don't see how these map to the keymaps (may be looking in the wrong place). No matter which I select the behaviour is the same.

Would be nice if the default Siri remote setup worked for CEC also but I'm not sure how all of this ties together currently.

**Comment by sy6sy2** [comment]
Hum you not have the last version. The settings that you are seeing are settings for macOS.
You should see the same settings in input as the ones in my screenshot above.

**Comment by phunkyfish** [comment]
Ah, ok. Thanks. Let me upgrade first and see how the remote works for CEC.

I'm still keen to know what didn't work between the two SiriRemote configs above though.
